bson/src/main/org/bson/Renderable.java

```diff
@@ -0,0 +1,20 @@
+package org.bson;
+
+import org.bson.codecs.configuration.CodecRegistry;
+
+/**
+ * An interface for types that are able to render themselves into a {@code BsonDocument}.
+ *
+ * @since 3.0
+ */
+public interface Renderable {
+    /**
+     * Render the filter into a BsonDocument.
+     *
+     * @param documentClass the document class in scope for the collection
+     * @param codecRegistry the codec registry
+     * @param <TDocument> the type of the document class
+     * @return the filter as a BsonDocument
+     */
+    <TDocument> BsonDocument render(Class<TDocument> documentClass, CodecRegistry codecRegistry);
```

Comment:
Is `documentClass` needed? I can't see where its actually used by a `Renderable`.